File: e2_CountTheHiddenSequences/csharp/Solution.cs

using System;

public class Solution {
    public int NumberOfArrays(int[] differences, int lower, int upper) {
        long minDiffSum = 0, maxDiffSum = 0, diffSums = 0;

        foreach (int diff in differences) {
            diffSums += diff;
            minDiffSum = Math.Min(minDiffSum, diffSums);
            maxDiffSum = Math.Max(maxDiffSum, diffSums);
        }

        long minInitial = lower - minDiffSum;
        long maxInitial = upper - maxDiffSum;

        return (int)Math.Max(0, maxInitial - minInitial + 1);
    }
}

File: e2_CountTheHiddenSequences/typescript/Solution.ts
class Solution {
    numberOfArrays(differences: number[], lower: number, upper: number): number {
        let minDiffSum: number = 0;
        let maxDiffSum: number = 0;
        let diffSums: number = 0;

        for (const diff of differences) {
            diffSums += diff;
            minDiffSum = Math.min(minDiffSum, diffSums);
            maxDiffSum = Math.max(maxDiffSum, diffSums);
        }

        const minInitial: number = lower - maxDiffSum;
        const maxInitial: number = upper - minDiffSum;

        return Math.max(0, maxInitial - minInitial + 1);
    }
}

export { Solution };